nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/AbstractAMQPProcessor.java

```diff
@@ -190,33 +197,50 @@
     public final void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
         AMQPResource<T> resource = resourceQueue.poll();
         if (resource == null) {
-            resource = createResource(context);
+            try {
+                resource = createResource(context);
+            } catch (Exception e) {
+                getLogger().error("Failed to initialize AMQP client", e);
+                context.yield();
+                return;
+            }
+        } else if (!resource.isAlive()) {
```

Comment:
With this we put checks in two places, _before_ and _during_ the AMQP communication. And the general problem is, the resource could "die" between the two.

It might be better to incorporate all resource health checks in the _during_ phase.

Reply:
"is alive" check moved to the consumer.